src/dmd/typesem.d

```diff
@@ -237,9 +244,26 @@ private void resolveHelper(TypeQualified mt, const ref Loc loc, Scope* sc, Dsymb
                 *pt = Type.terror;
                 return;
             }
+
+            void helper3()
+            {
+                Expression e;
+                VarDeclaration v = s.isVarDeclaration();
```

Comment:
I am used to isSomeType() returning a boolean, which is not the case here... But I can't suggest a better name than getVarDeclaration() so I guess isVarDeclaration() is as good as ahything else... Same goes for all similar methods.

Reply:
Yes, the `isSomeType` methods in DMD aren't the typical `isSomeType`, but we'd have to make a multi-thousand line PR to DMD to remedy that, and @WalterBright  has forbade that.  (See https://github.com/dlang/dmd/pull/7747).  That PR actually prompted @andralex to give me a call on my personal phone.  Yes, the great Andrei Alexandrescu called me... to scold me, that is.  One of the low moments in my D career, I must say.

Reply:
@JinShil well as I remember it it was more for planning future work than that PR. Sorry I came across as negative!

Reply:
I know you were just trying to help me understand your expectations.  No offense was taken.